src/extractor/edge_based_graph_factory.cpp

```diff
@@ -12,15 +13,42 @@
 
 #include <boost/assert.hpp>
 
+#include <algorithm>
+#include <cmath>
 #include <fstream>
 #include <iomanip>
 #include <limits>
+#include <sstream>
+#include <string>
 
 namespace osrm
 {
 namespace extractor
 {
 
+namespace
+{ // configuration parameters
+const bool constexpr INVERT = true;
+const bool constexpr RESOLVE_TO_RIGHT = true;
+const bool constexpr RESOLVE_TO_LEFT = false;
+
+// what angle is interpreted as going straight
+const double constexpr STRAIGHT_ANGLE = 180.;
+const double constexpr U_TURN_ANGLE = 0.;
+// if a turn deviates this much from going straight, it will be kept straight
+const double constexpr MAXIMAL_ALLOWED_NO_TURN_DEVIATION = 2.;
+// angle that lies between two nearly indistinguishable roads
+const double constexpr NARROW_TURN_ANGLE = 25.;
+// angle between roads that classifies them as fully distinct
+const double constexpr MINIMAL_DISTINCTION_ANGLE = 85.;
+// angle difference that can be classified as straight, if its the only narrow turn
+const double constexpr FUZZY_STRAIGHT_ANGLE = 15.;
+
+// Configuration to find representative candidate for turn angle calculations
+const double constexpr MINIMAL_SEGMENT_LENGTH = 1.;
+const double constexpr DESIRED_SEGMENT_LENGTH = 10.;
+} // namespace
```

Comment:
You don't need the anonymous namespace for `const constexpr`, they are only visible in that TU anyway.